pages/k8s/azure-cloud-provider.md

```diff
@@ -0,0 +1,203 @@
+  </div>
+</div>
+
+<div class="p-notification--caution is-inline">
+  <div class="p-notification__content">
+    <span class="p-notification__title">Resource usage:</span>
+    <p class="p-notification__message">By relating to this charm, other charms can directly allocate resources, such
+    as managed disks and load balancers, which could lead to cloud charges and
+    count against quotas. Because these resources are not managed by Juju, they
+    will not be automatically deleted when the models or applications are
+    destroyed, nor will they show up in Juju's status or GUI. It is therefore up
+    to the operator to manually delete these resources when they are no longer
+    needed, using the Azure management website or API.</p>
+  </div>
+</div>
+
+## Storage
+
+This section describes creating a busybox pod with a persistent volume claim
+backed by
+Azure's Disk Storage.
+
+### 1. Use the provided storage class using the `azuredisk-csi-provisioner` provisioner:
+
+```bash
+kubectl describe sc csi-azure-default
+```
+
+### 2. Create a persistent volume claim using that storage class:
+
+```bash
+kubectl create -f - <<EOY
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: testclaim
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 100Mi
+  storageClassName: csi-azure-default
+EOY
+```
+
+### 3. Create the busybox pod with a volume using that PVC:
+
+```bash
+kubectl create -f - <<EOY
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox
+  namespace: default
+spec:
+  containers:
+    - image: busybox
+      command:
+        - sleep
+        - "3600"
+      imagePullPolicy: IfNotPresent
+      name: busybox
+      volumeMounts:
+        - mountPath: "/pv"
+          name: testvolume
+  restartPolicy: Always
+  volumes:
+    - name: testvolume
+      persistentVolumeClaim:
+        claimName: testclaim
+EOY
+```
+
+Charmed Kubernetes can make use of additional types of storage - for more
+information see the [storage documentation][storage].
+
+## Azure load-balancers for services
+
+The following commands start the 'hello-world' pod behind an Azure-backed
+load-balancer.
+
+Here are the commands for Kubernetes 1.25+ and above as the kubectl run command was deprecated:
+
+```bash
+# Kubernetes 1.25+
+kubectl create deployment hello-world --image=gcr.io/google-samples/hello-app:1.0  --port=8080
+kubectl expose deployment hello-world --type=LoadBalancer --name=hello
+watch kubectl get svc -o wide --selector=app=hello-world
+```
+
+You can then verify this works by loading the described IP address (on port
+8080!) in a browser.
+
+For more configuration options and details of the permissions which the integrator uses,
+please see the [charm page][azure-cloud-provider].
+
+## Azure load-balancers for the control plane
+
+With revision 1015 and later of the `kubernetes-control-plane` charm, Charmed
```

Comment:
Does `revision 1015` mean anything useful after we moved to tracks in charmhub?  Possibly better to say `With 1.25/stable and later`?